torchvision/prototype/models/quantization/shufflenetv2.py

```diff
@@ -87,47 +88,43 @@ class ShuffleNet_V2_X1_0_QuantizedWeights(WeightsEnum):
     default = ImageNet1K_FBGEMM_V1
 
 
+@handle_legacy_interface(
+    weights=(
+        "pretrained",
+        lambda kwargs: ShuffleNet_V2_X0_5_QuantizedWeights.ImageNet1K_FBGEMM_V1
+        if kwargs.get("quantize", False)
+        else ShuffleNet_V2_X0_5_Weights.ImageNet1K_V1,
```

Comment:
Perfect. This mechanism should also support the `KeypointRCNN_ResNet50_FPN_Weights.Coco_Legacy` case, but check it does. I would also check all other special cases I marked on an earlier PR before starting replacing it form everywhere to ensure your solution works fine. I think it does.